models/stdp_dopamine_synapse.cpp

```diff
@@ -82,10 +84,16 @@ STDPDopaCommonProperties::set_status( const DictionaryDatum& d, ConnectorModel&
   long vtnode_id;
   if ( updateValue< long >( d, names::vt, vtnode_id ) )
   {
-    const thread tid = kernel().vp_manager.get_thread_id();
-    Node* vt = kernel().node_manager.get_node_or_proxy( vtnode_id, tid );
-    vt_ = dynamic_cast< volume_transmitter* >( vt );
-    if ( not vt_ )
+    std::string model_name = ClassToString( volume_transmitter );
```

Comment:
Why is the macro necessary; why can't we just use the const string literal ``"volume_transmitter"``?

Reply:
Now that you say it, using the macro is actually incorrect here, as the name of the class does not have to match the name of the model. This works as a workaround, but what we would actually need would be a way to retrieve the type_id of a model from the type, so a function of the form:
`index ModelManager::get_type_id<volume_transmitter>();`
However, in this case, we could just assume that `volume_transmitter` is also registered as `"volume_transmitter"`, even though this isn't the cleanest way and might break the code if the registration was to be changed.